doc/lsst.pipe.base/creating-a-task.rst

```diff
@@ -0,0 +1,350 @@
+  Your task will be much easier to use if the default configuration usually suffices.
+
+- Document each field of the configuration carefully.
+  Pretend you don't know anything about your task and ask yourself what you would need to be told to change the parameter.
+  What does the parameter do and why might you change it?
+  What units does it have?
+
+- Subtasks are specified in your configuration as fields of type `lsst.pex.config.ConfigurableField` or (less commonly) `lsst.pex.config.RegistryField`.
+  This allows subtasks to be :doc:`retargeted <command-line-task-retargeting-howto>` (replaced with a variant subtask).
+  For more information see :ref:`creating-a-task-subtasks`.
+
+  `~lsst.pipe.tasks.exampleStatsTasks.ExampleSigmaClippedStatsTask` uses configuration class `~lsst.pipe.tasks.exampleStatsTasks.ExampleSigmaClippedStatsConfig`:
+
+  .. code-block:: python
+
+     class ExampleSigmaClippedStatsConfig(pexConfig.Config):
+         """Configuration for ExampleSigmaClippedStatsTask.
+         """
+         badMaskPlanes = pexConfig.ListField(
+             dtype=str,
+             doc="Mask planes that, if set, indicate the associated pixel should "
+             "not be included when the calculating statistics.",
+             default=("EDGE",),
+         )
+         numSigmaClip = pexConfig.Field(
+             doc="number of sigmas at which to clip data",
+             dtype=float,
+             default=3.0,
+         )
+         numIter = pexConfig.Field(
+             doc="number of iterations of sigma clipping",
+             dtype=int,
+             default=2,
+         )
+
+The configuration class is specified as ``ExampleSigmaClippedStatsTask`` class variable ``ConfigClass``, as described in :ref:`creating-a-task-class-variables`.
+
+Here is the config for ``ExampleCmdLineTask``, a task that calls one subtask named ``stats``; notice the `lsst.pex.config.ConfigurableField`:
+
+.. code-block:: python
+
+   class ExampleCmdLineConfig(pexConfig.Config):
+       """Configuration for ExampleCmdLineTask.
+       """
+       stats = pexConfig.ConfigurableField(
+           doc="Subtask to compute statistics of an image",
+           target=ExampleSigmaClippedStatsTask,
+       )
+       doFail = pexConfig.Field(
+           doc="Raise an lsst.base.TaskError exception when processing each image? "
+           + "This allows one to see the effects of the --doraise command line flag",
+           dtype=bool,
+           default=False,
+       )
+
+.. _creating-a-task-class-variables:
+
+Class variables
+===============
+
+Tasks require several class variables to function:
+
+- ``ConfigClass``: the :ref:`configuration class <creating-a-task-configuration>` used by the task.
+
+- ``_DefaultName``: a string used as the default name for the task.
+  This is required for a command-line task (`~pipe.base.cmdLineTask.CmdLineTask`), and strongly recommended for other tasks because it makes them easier to construct for unit tests.
```

Comment:
Why is it not required?  Here's an example of a place that while writing the docs we should also file the tickets to fix obvious problems.  If there is a reason, on the other hand, it needs to be documented here.